app/models/report.rb

```diff
@@ -27,17 +29,19 @@ def created_on
     created_at.to_date
   end
 
-  def update_mentions
+  def save_mentions
     uri_ids = extract_uri_ids
     mentioned_reports.destroy_all
-    save_mentions(uri_ids)
+    create_mentions(uri_ids)
   end
 
-  def save_mentions(uri_ids)
-    uri_ids.each do |uri_id|
-      next unless other_report?(uri_id)
+  def create_mentions(uri_ids)
+    transaction do
+      uri_ids.each do |uri_id|
+        next unless other_report?(uri_id)
```

Comment:
```suggestion
        next if uri_id.to_i == id
```

これでも「自分に対する言及だったら処理をスキップ」と読めるのでメソッド化はしなくても大丈夫かもしれません。